hs_collection_resource/templates/pages/collectionresource.html

```diff
@@ -39,64 +39,67 @@ <h3>Collection Contents </h3>
+                                {% with res|user_permission:request.user.pk as upermission %}
+                                    <tr id="{{ res.short_id }}">
+                                        <td>
+                                            <input class="row-selector" type="checkbox" id="{{ res.short_id }}">
+                                        </td>
+                                        <td>
+                                            {% if upermission|lower == "none" %}
+                                                <strong>{{ res.metadata.title }}</strong>
+                                            {% else %}
+                                                <strong><a href="{{ res.get_absolute_url }}"
+                                                           target="_blank">{{ res.metadata.title }}</a></strong>
+                                            {% endif %}
+                                        </td>
+                                        <td>
+                                            {{ res.resource_type }}
+                                        </td>
+                                        <td>
+                                            {% with raccess.owners.all|first as owner %}
+                                                {% if owner.first_name %}
+                                                    <strong><a href='/user/{{ owner.pk }}/'
+                                                               target="_blank">{{ owner.first_name }} {{ owner.last_name }}</a></strong>
+                                                {% else %}
+                                                    <strong><a href='/user/{{ owner.pk }}/'
+                                                               target="_blank">{{ owner.username }}</a></strong>
+                                                {% endif %}
+                                            {% endwith %}
+                                        </td>
+                                        <td>
+                                            {% if raccess.published %}
+                                                Published
+                                            {% elif raccess.public %}
+                                                Public
+                                            {% elif raccess.discoverable %}
+                                                Discoverable
+                                            {% else %}
+                                                Private
+                                            {% endif %}
 
-                                    {% if res.raccess.shareable %}
-                                         & Shareable
-                                    {% endif %}
-                                </td>
-                                <td>
-                                    {% if res|user_permission:request.user.pk|lower == "none" %}
```

Comment:
This block seemed to be the major bottleneck.  the user_permission tag is slow and was called at least 2-3 times per resource.  I dropped it to once per resource.